src/js/mapMerge.js

```diff
@@ -0,0 +1,33 @@
+"use strict";
+var fluid = require("infusion");
+fluid.registerNamespace("fluid.lintAll");
+
+fluid.lintAll.mapMerge = function (baseArray, toMergeArray) {
```

Comment:
Some JSDocs and strategy comments would be helpful here - what does this function achieve?

Reply:
I have added JSDocs to describe what the trio of functions does.  The previous docs were inaccurate and described the old behaviour.  I have updated those as well.